api/src/services/organizations_v1/list_legacy_users.py

```diff
@@ -0,0 +1,225 @@
+    db_session: db.Session,
+    user: User,
+    organization_id: UUID,
+    request_data: dict,
+) -> tuple[list[dict], PaginationInfo]:
+    """
+    List legacy users from Oracle staging tables that can be invited to the organization.
+
+    Uses SQL window function for efficient deduplication by email.
+    Computes status based on membership and invitation state.
+    Filters to organization applicant profile type (profile_type_id=4).
+
+    Args:
+        db_session: Database session
+        user: Authenticated user making the request
+        organization_id: Organization ID to list users for
+        request_data: Request body containing pagination parameters and filters
+
+    Returns:
+        Tuple of (list of legacy user dicts with status, pagination info)
+
+    Raises:
+        403: User lacks MANAGE_ORG_MEMBERS privilege
+        404: Organization not found
+        400: Organization does not have a UEI
+    """
+    # Get organization and verify access
+    organization = get_organization_and_verify_access(db_session, user, organization_id)
+
+    # Get organization UEI from SAM.gov entity
+    uei = organization.sam_gov_entity.uei if organization.sam_gov_entity else None
+    if not uei:
+        raise_flask_error(400, "Organization does not have a UEI")
+
+    # Parse pagination parameters and filters
+    params = LegacyUserListParams.model_validate(request_data)
+
+    # Build subqueries to check membership and invitation status
+
+    # Subquery to check if user is a member
+    is_member_subquery = (
+        exists(
+            select(1)
+            .select_from(OrganizationUser)
+            .join(User, OrganizationUser.user_id == User.user_id)
+            .join(LinkExternalUser, User.user_id == LinkExternalUser.user_id)
+            .where(
+                OrganizationUser.organization_id == organization_id,
+                func.lower(LinkExternalUser.email) == func.lower(VuserAccount.email),
+            )
+        )
+    ).label("is_member")
+
+    # Subquery to check if user has pending invitation
+    has_pending_invitation_subquery = (
+        exists(
+            select(1)
+            .select_from(OrganizationInvitation)
+            .where(
+                OrganizationInvitation.organization_id == organization_id,
+                func.lower(OrganizationInvitation.invitee_email) == func.lower(VuserAccount.email),
+                OrganizationInvitation.accepted_at.is_(None),
+                OrganizationInvitation.rejected_at.is_(None),
+                OrganizationInvitation.expires_at > func.now(),
+            )
+        )
+    ).label("has_pending_invitation")
+
+    # Build deduplicated query with status computation using window function
+    # Use ROW_NUMBER() to rank duplicates by created_date (most recent first)
+    ranked_subquery = (
+        select(
+            VuserAccount.user_account_id,
+            VuserAccount.email,
+            VuserAccount.full_name,
```

Comment:
Should we use the first/last name fields instead? We might want to check, when grants.gov tables have these duplicative columns, usually one of them is missing some data.